File: Go/exercicio7.go
package main

import "fmt"

type Professor struct {
    Nome string
}

type Escola struct {
    Professores []Professor
}

func (e *Escola) AdicionarProfessor(p Professor) {
    e.Professores = append(e.Professores, p)
}

func main() {
    escola := Escola{}
    escola.AdicionarProfessor(Professor{"Maria"})
    escola.AdicionarProfessor(Professor{"José"})

    for _, professor := range escola.Professores {
        fmt.Println(professor.Nome)
    }
}

File: Python/exercicio7.py
class Escola:
    def __init__(self):
        self.professores = []

    def adicionar_professor(self, professor):
        self.professores.append(professor)

class Professor:
    def __init__(self, nome):
        self.nome = nome

escola = Escola()
prof1 = Professor("Maria")
prof2 = Professor("José")

escola.adicionar_professor(prof1)
escola.adicionar_professor(prof2)

for professor in escola.professores:
    print(professor.nome)